src/DistributedMetadata/DDLService.cpp

```diff
@@ -415,10 +435,19 @@ void DDLService::mutateTable(IDistributedWriteAheadLog::RecordPtr record, const
     String user = block.getByName("user").column->getDataAt(0).toString();
     String payload = block.getByName("payload").column->getDataAt(0).toString();
 
-    std::vector<Poco::URI> target_hosts{toURIs(
-        placement.placed(database, table),
-        fmt::format(DDL_TABLE_PATCH_API_PATH_FMT, database, getURIEndpoint(record->headers), table),
-        http_port)};
+    std::vector<Poco::URI> target_hosts;
```

Comment:
Can we refactor this out as a separate function and refactor out a `getTableApiPath(...)` just as `getColumnApiPath`. In this way, the code will keep consistent.

```
String getTableApiPath(...)
{
...
}

std::vector<Poco::URI> getTargetURIs(const std::unordered_map<String, String> & headers) const
{
    column_ddl_request = headers.contains("column")
    if (column_ddl_request)
    {
        method = headers.at("query_method");
        target_hosts = toURIs(
            placement.placed(database, table), 
            getColumnApiPath(headers, database, table, method), 
            http_port);
    }
    else
    {
       /// Table DDL request
        target_hosts = toURIs(
            placement.placed(database, table),
            getTableApiPath(headers, database, table, method),
            http_port);
    }
}

```

Reply:
If we introduce more OpCodes, probably we don't need use the key in the headers to differentiate if it is column operation or table operation. Please see some comments below.

Reply:
fixed